### PR TITLE
Add `pip install --constraint` test coverage

### PR DESCRIPTION
**Pull request by zanieb**
Exploring behavior reported in https://github.com/astral-sh/uv/issues/1332

### CODE REVIEW

File: crates/uv/tests/pip_install.rs [hunk 1]
```diff
@@ -894,3 +894,61 @@ fn no_deps() {
...
+    uv_snapshot!(command(&context)
+            .arg("-r")
+            .arg("requirements.txt"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Audited 0 packages in [TIME]
+    "###
```
Comment (charliermarsh) [inline crates/uv/tests/pip_install.rs:950]: This is the bug, right?

Comment (zanieb) [reply crates/uv/tests/pip_install.rs:950]: It sure looks wrong to me.

Comment (zanieb) [reply crates/uv/tests/pip_install.rs:1068]: Looking into this.. really weird silent failure in parsing?

Comment (zanieb) [reply crates/uv/tests/pip_install.rs:1072]: Bad test — looks fine https://github.com/astral-sh/uv/pull/1334/commits/81c466c16c556dce872223e8ed8927d85759b7b0

**Pull request merged**